[JavaScriptCore] strstr->strchr when string is only one letter long

LumioseSil opened the pull request:
#### 5d0b9adc2945e30d8bb92a6ebe31b0c745b4c667
<pre>
[JavaScriptCore] strstr-&gt;strchr when string is only one letter long
<a href="https://bugs.webkit.org/show_bug.cgi?id=267351">https://bugs.webkit.org/show_bug.cgi?id=267351</a>

Reviewed by Michael Catanzaro.

Instead of having to find a whole string, we can just find the char we
need.

* Source/JavaScriptCore/runtime/Options.cpp: strstr-&gt;strchr
(JSC::Options::setOptions): Ditto.
* Source/JavaScriptCore/tools/FunctionOverrides.cpp: Ditto.
(JSC::parseClause): Ditto.

Canonical link: <a href="https://commits.webkit.org/272859@main">https://commits.webkit.org/272859@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e448f497f12a9424b0edebb25de767f8e9592d6e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33418 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12190 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/35332 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36045 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30366 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/34390 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14540 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9342 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/29497 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33891 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10280 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29835 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8973 "Found 1 new API test failure: /WPE/TestResources:/webkit/WebKitWebView/sync-request-on-max-conns (failure)") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/9099 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/29825 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/37375 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/28548 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30346 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/30173 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35221 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/33412 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9224 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/7168 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33089 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10977 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/39885 "Built successfully") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/7728 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9810 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8366 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9949 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitInitialize.cpp:
```diff
@@ -53,7 +53,7 @@ static void initializeRemoteInspectorServer()
 
         GUniquePtr<char> inspectorAddress(g_strdup(address));
-        char* portPtr = g_strrstr(inspectorAddress.get(), ":");
+        char* portPtr = g_strrchr(inspectorAddress.get(), ':');
```
There is no `g_strrchr()` (which is why this doesn't build). You can change it to use `strrchr()` though.

mcatanzaro replied:
That said, the title of your commit message says "JavaScriptCore" and you are outside of it here, so this change seems misplaced regardless.

LumioseSil committed:
[JavaScriptCore] strstr->strchr when string is only one letter long
https://bugs.webkit.org/show_bug.cgi?id=267351

Reviewed by Michael Catanzaro.

Instead of having to find a whole string, we can just find the char we
need.

* Source/JavaScriptCore/runtime/Options.cpp: strstr->strchr
(JSC::Options::setOptions): Ditto.
* Source/JavaScriptCore/tools/FunctionOverrides.cpp: Ditto.
(JSC::parseClause): Ditto.

Canonical link: https://commits.webkit.org/272859@main
